iroha/src/block_sync.rs

```diff
@@ -263,19 +263,16 @@ pub mod message {
 
 /// This module contains all configuration related logic.
 pub mod config {
-    use std::env;
+    use iroha_config::derive::Configurable;
+    use serde::{Deserialize, Serialize};
 
-    use iroha_error::{Result, WrapErr};
-    use serde::Deserialize;
-
-    const BATCH_SIZE: &str = "BLOCK_SYNC_BATCH_SIZE";
-    const GOSSIP_PERIOD_MS: &str = "BLOCK_SYNC_GOSSIP_PERIOD_MS";
     const DEFAULT_BATCH_SIZE: u64 = 4;
     const DEFAULT_GOSSIP_PERIOD_MS: u64 = 10000;
 
     /// Configuration for `BlockSynchronizer`.
-    #[derive(Copy, Clone, Deserialize, Debug)]
+    #[derive(Copy, Clone, Deserialize, Serialize, Debug, Configurable)]
     #[serde(rename_all = "UPPERCASE")]
+    #[config(env_prefix = "BLOCK_SYNC_")]
```

Comment:
All the prefixes will end up with `_`. Maybe we should move this separator into impl.

Reply:
IMO here it better be explicit than implicit